docs/design/ovs-pipeline.md

```diff
@@ -183,1015 +244,1228 @@ spec:
-a Pod will not be forwarded back to OVS pipeline as the destination IP is the Pod's
-own IP, then the connection is interrupted because the conntrack state is only stored
-in OVS ct zone, not in the Pod. With SNAT, the destination IP will be the virtual
-hairpin IP and forwarded back to OVS pipeline. Note that, bit 18 in NXM_NX_REG0 is
-set to 0x1, and it is consumed in [L2ForwardingOutTable] to output the packet
-to the port on which it was received with action `IN_PORT`.
+- Flows 1-2 matches packets from the sample Antrea-native NetworkPolicy ingress rule with action `Allow`.
+  - Match conditions `ct_state=+new` or `ct_state=-new` are to match packets with state `NEW` or not `NEW`.
+  - Match condition `ct_label=0x6/0xffffffff` is to match `IngressRuleCTLabel` with value 6, which is loaded in table
+    [AntreaPolicyIngressRule], flow 8.
+  - Action `goto_table:ConntrackCommit` is to forward packets to table [ConntrackCommit] since the action of the sample
+    rule is `Allow`.
+- Flows 3-4 matches packets from the sample Kubernetes NetworkPolicy ingress rule with action `Allow`.
+  - Match conditions `ct_state=+new` or `ct_state=-new` are to match packets with state `NEW` or not `NEW`.
+  - Match condition `ct_label=0x7/0xffffffff` is to match `EgressRuleCTLabel` with value 7, which is loaded in table
+    [IngressRule], flow 6.
+  - Action `goto_table:ConntrackCommit` is the same as flows 1-2.
+- Flow 5 matches packets of the sample Antrea-native NetworkPolicy ingress rule with action `Drop` or `Reject`, then
+  drop them.
+  - Match condition `reg0=0x400/0x400` is to match `APDenyRegMark`, which is loaded in table [AntreaPolicyIngressRule],
+    flow 12, indicating that the packets should be denied.
+  - Match condition `reg3=0x4` is to match `APConjIDField`, which is loaded in table [AntreaPolicyIngressRule], flow 12
+  - Action `drop` is to drop packets.
+- Flow 6 is the auto-generated flow.
+
+### ConntrackCommit
+
+This table is in charge of committing all new non-Service connections.
 
-### L2ForwardingOutTable (110)
-
-It is a simple table and if you dump the flows for this table, you should only
-see 2 flows:
+If you dump the flows for this table, you should see the following:
 
 ```text
-1. table=110, priority=200,ip,reg0=0x10000/0x10000 actions=output:NXM_NX_REG1[]
-2. table=110, priority=0, actions=drop
+1. table=ConntrackCommit, priority=200,ct_state=+new+trk-snat,ct_mark=0/0x10,ip actions=ct(commit,table=Output,zone=65520,exec(move:NXM_NX_REG0[0..3]->NXM_NX_CT_MARK[0..3]))
+2. table=ConntrackCommit, priority=0 actions=goto_table:Output
 ```
 
-The first flow outputs all unicast packets to the correct port (the port was
-resolved by the "dmac" table, [L2ForwardingCalcTable]). IP packets for which
-[L2ForwardingCalcTable] did not set bit 16 of NXM_NX_REG0 will be dropped.
+- Flow 1 is to match new non-Service connections and commit them to the connection tracking module.
```

Comment:
Why we don't use "-dnat" state in the flows in this table?

Reply:
We have a ct mark to distinguish Service packets. The ct mark is sufficient.